scene/gui/tree.cpp

```diff
@@ -96,10 +96,8 @@ void TreeItem::_change_tree(Tree *p_tree) {
 		return;
 	}
 
-	TreeItem *c = first_child;
-	while (c) {
+	for (auto& c : children) {
```

Comment:
`auto` is [not allowed](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#auto-keyword)